### PR TITLE
Update dependency com.azure:azure-identity to 1.13.0, com.microsoft.azure:msal4j to 1.15.1, com.microsoft.azure:msal4j-persistence-extension to 1.3.0

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Update dependency com.azure:azure-identity to 1.12.2

Update dependency com.microsoft.azure:msal4j to 1.15.1

Update dependency com.microsoft.azure:msal4j-persistence-extension to 1.13.0

Replacement for the mend PR which did not update SHAs or CHANGELOG: https://github.com/opensearch-project/OpenSearch/pull/14494

### Check List
- [X] Functionality includes testing.
- [X] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/repository-azure/build.gradle [hunk 1]
```diff
@@ -57,11 +57,11 @@ dependencies {
   api "io.netty:netty-transport-native-unix-common:${versions.netty}"
   implementation project(':modules:transport-netty4')
   api 'com.azure:azure-storage-blob:12.23.0'
-  api 'com.azure:azure-identity:1.11.4'
+  api 'com.azure:azure-identity:1.12.2'
```
Comment (reta) [inline plugins/repository-azure/build.gradle:60]:
```suggestion
  api 'com.azure:azure-identity:1.13.0'
```

File: plugins/repository-azure/build.gradle [hunk 2]
```diff
@@ -57,11 +57,11 @@ dependencies {
...
   // Start of transitive dependencies for azure-identity
   api 'com.microsoft.azure:msal4j-persistence-extension:1.2.0'
```
Comment (reta) [inline plugins/repository-azure/build.gradle:62]:
```suggestion
  api 'com.microsoft.azure:msal4j-persistence-extension:1.3.0'
```

**Comment by reta** [comment]
Thanks @cwperks , for some reasons Mend completely wiped out my changes https://github.com/opensearch-project/OpenSearch/commit/4ed2b988233a264b1bca756d0d0ad71a31899ff6 from its pull request

File: plugins/repository-azure/build.gradle (hunk 1 above)
Comment (cwperks) [reply plugins/repository-azure/build.gradle]: Updated

File: plugins/repository-azure/build.gradle (hunk 2 above)
Comment (cwperks) [reply plugins/repository-azure/build.gradle]: Updated

**Comment by cwperks** [comment]
Thanks for the review @reta! Mend must have overridden your commit.

**Review by reta — APPROVED** [review]

**Pull request merged**